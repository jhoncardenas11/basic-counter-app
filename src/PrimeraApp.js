// Funcional components
import React from "react";
import propTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitle }) => {

    return (
        <>
            {/* <pre> { JSON.stringify(saludo, null, 3) } </pre> */}
            <h1>{ saludo }</h1>
            <p>{ subtitle }</p>
        </>
    )

}

PrimeraApp.propTypes = {
    saludo: propTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle: 'soy un subtitulo'
}

export default PrimeraApp;
