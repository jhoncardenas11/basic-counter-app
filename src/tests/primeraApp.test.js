import { shallow } from 'enzyme';
// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe('pruebas en <PrimeraApp />', () => {

    // con test normal jest.
    // test('Debe de mostrar el mensaje "Hola mundo!"', () => {

    //     const saludo = "Hola mundo!";
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect( getByText(saludo) ).toBeInTheDocument();

    // })

    // Con enzyme
    test('Debe de mostrar < PrimeraApp /> correctamente', () => {

        const saludo = "Hola mundo!"
        const wrapper = shallow( <PrimeraApp saludo={saludo}/> );

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe enviar el subtitulo enviado por props', () => {

        const saludo = "Hola mundo!"
        const subtitulo = "Soy un subtitulo";
        const wrapper = shallow( <
            PrimeraApp saludo={saludo}
            subtitle={subtitulo}
        /> );

        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitulo);

    })

});
