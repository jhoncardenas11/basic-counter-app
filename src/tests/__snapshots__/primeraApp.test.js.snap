// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <PrimeraApp /> Debe de mostrar < PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola mundo!
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
